irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -866,16 +868,47 @@ namespace iroha {
       using QueryTuple =
           QueryType<shared_model::interface::types::AccountIdType,
                     shared_model::interface::types::AssetIdType,
-                    std::string>;
+                    std::string,
+                    size_t>;
       using PermissionTuple = boost::tuple<int>;
 
-      auto cmd = (boost::format(R"(WITH has_perms AS (%s),
-      t AS (
-          SELECT * FROM account_has_asset
-          WHERE account_id = :account_id
+      // get the assets
+      auto cmd = (boost::format(R"(
+      with has_perms as (%s),
+      all_data as (
+          select row_number() over () rn, *
```

Comment:
> Is it possible to use `LIMIT M, N` instead of artificially introduced row_number for partial selection?

I could not create a better query without row numbers. 

> What does `()` after `over` mean?

That's such a syntax for [window functions](https://www.postgresql.org/docs/9.5/functions-window.html). AFAIK you can put some specifiers in them like alternative ordering.

Reply:
() looks very strange for me and I got no answer. Anyway, I gonna approve that if it works.